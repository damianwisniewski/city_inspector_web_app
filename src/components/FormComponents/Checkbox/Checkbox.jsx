import React from 'react'
import PropTypes from 'prop-types'

const Checkbox = ({ id, children, ...rest }) => (
	<div className='input-wrapper'>
		<input {...rest} type='checkbox' className='input-wrapper__field' id={id} />
		<label className='input-wrapper__label' htmlFor={id}>
			{children}
		</label>
	</div>
)

Checkbox.propTypes = {
	id: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
}

export default Checkbox
