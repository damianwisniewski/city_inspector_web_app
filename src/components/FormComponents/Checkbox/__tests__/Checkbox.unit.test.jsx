import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Checkbox from '../Checkbox'

configure({ adapter: new Adapter() })

describe('<Checkbox />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let CheckoutElement
	const CheckoutText = 'ExampleText'
	const id = 'mockId'

	test('should render correctly Checkbox', () => {
		CheckoutElement = shallow(<Checkbox id={id} label={CheckoutText} />)
		expect(toJson(CheckoutElement)).toMatchSnapshot()
	})
})
