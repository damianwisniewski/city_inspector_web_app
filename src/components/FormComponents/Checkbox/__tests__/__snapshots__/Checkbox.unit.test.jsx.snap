// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render correctly Checkbox 1`] = `
<div
  className="input-wrapper"
>
  <input
    className="input-wrapper__field"
    id="mockId"
    type="checkbox"
  />
  <label
    className="input-wrapper__label"
    htmlFor="mockId"
  >
    ExampleText
  </label>
</div>
`;
